WalletWasabi.Gui/Controls/WalletExplorer/CoinJoinTabViewModel.cs

```diff
@@ -510,5 +509,46 @@ private TargetPrivacy TargetPrivacy
 		public ReactiveCommand PrivacyFineCommand { get; }
 		public ReactiveCommand PrivacyStrongCommand { get; }
 		public ReactiveCommand TargetButtonCommand { get; }
+
+		#region IDisposable Support
+
+		private volatile bool _disposedValue = false; // To detect redundant calls
+
+		protected virtual void Dispose(bool disposing)
+		{
+			if (!_disposedValue)
+			{
+				if (disposing)
+				{
+					if (Global.ChaumianClient != null)
+					{
+						Global.ChaumianClient.CoinQueued -= ChaumianClient_CoinQueued;
+						Global.ChaumianClient.CoinDequeued -= ChaumianClient_CoinDequeued;
+						Global.ChaumianClient.StateUpdated -= ChaumianClient_StateUpdated;
+					}
+
+					if (_coinsList != null)
+					{
+						_coinsList.DequeueCoinsPressed -= CoinsList_DequeueCoinsPressedAsync;
+					}
+
+					_disposables?.Dispose();
+					CoinsList?.Dispose();
+				}
+
+				CoinsList = null;
+				_disposedValue = true;
```

Comment:
Why are these outside of `if (disposing)`?

Reply:
I found the answer. I always forgot about this. I'll try to keep it in mind.

Reply:
Followed the dispose pattern. **TODO: set large fields to null.**

```
		protected virtual void Dispose(bool disposing)
		{
			if (!disposedValue)
			{
				if (disposing)
				{
					// TODO: dispose managed state (managed objects).
				}

				// TODO: free unmanaged resources (unmanaged objects) and override a finalizer below.
				// TODO: set large fields to null.

				disposedValue = true;
			}
		}
```

Reply:
Usually we are not doing this I just wanted to be precise. Later I found a bug because of this so I will use this in the future. (Object was disposed to early)